src/main/java/frc/robot/RobotContainer.java

```diff
@@ -35,252 +31,222 @@
+		.withPosition(18, 0).withSize(9, 4).withProperties(Map.of("min", 0.0, "max", 150.0)).getEntry()
+		.addListener(notif -> {
+			Constants.MOTOR_WARNING_TEMP = notif.value.getDouble(); 
+				}, EntryListenerFlags.kUpdate);
+				
+		configTab.add("Motor Shutoff Temp", Constants.MOTOR_SHUTOFF_TEMP).withWidget(BuiltInWidgets.kNumberSlider)
+		.withPosition(27, 0).withSize(9, 4).withProperties(Map.of("min", 0.0, "max", 150.0)).getEntry()
+		.addListener(notif -> {
+			Constants.MOTOR_SHUTOFF_TEMP = notif.value.getDouble(); 
+				}, EntryListenerFlags.kUpdate);
+				
+		// Drive Tabs 
+		driveTab.add("Gyro", drivetrain.getAHRS()).withWidget(BuiltInWidgets.kGyro).withPosition(0, 4).withSize(9, 10); 
+
+		// Driving Related Entries
+		driveReversedEntry = driveTab.add("Reversed", TeleopDrive.isReversed()).withWidget(BuiltInWidgets.kBooleanBox)
+				.withPosition(0, 0).withSize(4, 4).getEntry();
+		
+		precisionDriveEntry = driveTab.add("Precision", TeleopDrive.isPrecisionDrive()).withWidget(BuiltInWidgets.kBooleanBox)
+				.withPosition(4, 0).withSize(4, 4).getEntry();
+		
+		// Overheating Warnings 
+		drivetrain.getMonitorGroup().setOverheatShutoffCallback((motor, temp) -> {
+			if (!drivetrain.getOverheatShutoffOverride()) {
+				drivetrainMotorStatus.withProperties(Map.of("color when false", Constants.COLOR_MOTOR_SHUTOFF)).getEntry()
+						.setBoolean(false);
+			}
+			getLogger()
+					.logError("Drivetrain motor " + motor.getDeviceId() + " reached overheat shutoff limit at " + temp + "C!"); 
+		});
+
+		drivetrain.getMonitorGroup().setOverheatWarningCallback((motor, temp) -> {
+			if (!drivetrain.getOverheatShutoffOverride()) {
+				drivetrainMotorStatus.withProperties(Map.of("color when false", Constants.COLOR_MOTOR_WARNING))
+						.getEntry().setBoolean(false);
+			}
+			getLogger()
+					.logWarning("Drivetrain motor " + motor.getDeviceId() + " reached overheat warning at " + temp + " C!");
+		});
+
+		drivetrain.getMonitorGroup().setNormalTempCallback(() -> {
+			drivetrainMotorStatus.getEntry().setBoolean(true); 
+		});
+
+		shooterSubsystem.getMonitorGroup().setOverheatShutoffCallback((motor, temp) -> {
+			if (!shooterSubsystem.getOverheatShutoffOverride())
+				shooterMotorStatus.withProperties(Map.of("color when false", Constants.COLOR_MOTOR_SHUTOFF)).getEntry().setBoolean(false); 
+			getLogger().logError("Shooter motor " + motor.getDeviceId() + " reched overheat shutoff limit at " + temp + "C!"); 
+		});
+
+		shooterSubsystem.getMonitorGroup().setOverheatWarningCallback((motor, temp) -> {
+			if (!shooterSubsystem.getOverheatShutoffOverride())
+				shooterMotorStatus.withProperties(Map.of("color when false", Constants.COLOR_MOTOR_WARNING)).getEntry().setBoolean(false); 
+			getLogger().logWarning("Shooter motor " + motor.getDeviceId() + " reached overheat warning at " + temp + " C!"); 
+		});
+
+	}
+
+	private double shooterDistToSpeed(double dist, double endh) {
```

Comment:
If this is a shooter-specific function, I think we should move this to the commands or subsystems folder as RobotContainer.java usually does not include any subsystem-specific functions.

Reply:
Ok, that's probably a good idea.

Reply:
I did that